src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs

```diff
@@ -541,7 +544,7 @@ public int GetEra(string eraName)
             for (int i = 0; i < EraNames.Length; i++)
             {
                 // Compare the era name in a case-insensitive way for the appropriate culture.
-                if (m_eraNames[i].Length > 0)
+                if (m_eraNames![i].Length > 0)
```

Comment:
Briefly for my education why are `!` needed on the fields? Because they are xplicitly initialized before this method is called, and the compiler can't possibly know that?

Reply:
@danmosemsft `EraNames` ensures that field is initialized and compiler is not able to figure it out